airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -207,6 +207,88 @@ def __init__(self, service, project_id, use_legacy_sql=True):
         self.use_legacy_sql = use_legacy_sql
         self.running_job_id = None
 
+    def create_empty_table(self,
+                           project_id,
+                           dataset_id,
+                           table_id,
+                           schema_fields=None,
+                           time_partitioning={}
+                           ):
+        """
+        Creates a new, empty table in the dataset.
+
+        :param project_id: The project to create the table into.
+        :type project_id: str
+        :param dataset_id: The dataset to create the table into.
+        :type dataset_id: str
+        :param table_id: The Name of the table to be created.
+        :type table_id: str
+        :param schema_fields: The schema field list as defined here: ::
+            https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#resource
+        :type schema_fields: list
+        :param time_partitioning: configure optional time partitioning fields i.e.
+            partition by field, type and expiration as per API specifications.
+
+            .. seealso::
+            https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#timePartitioning
+        :type time_partitioning: dict
+
+        :return:
+        """
+        project_id = project_id if project_id is not None else self.project_id
+
+        table_resource = {
+            'tableReference': {
+                'tableId': table_id
+            }
+        }
+
+        if schema_fields:
+            table_resource['schema'] = {'fields': schema_fields}
+
+        if time_partitioning:
+            table_resource['timePartitioning'] = time_partitioning
+
+        tables_list_resp = self.service.tables().list(
+            projectId=project_id, datasetId=dataset_id).execute()
+        while True:
+            for table in tables_list_resp.get('tables', []):
+                if table['tableReference']['tableId'] == table_id:
+                    # If table exists raise error.
+                    raise AirflowException(
+                        'Already Exists: Table {}:{}.{}.'.format(
+                            project_id, dataset_id, table_id
+                        )
+                    )
+            # If there is a next page, we need to check the next page.
+            if 'nextPageToken' in tables_list_resp:
+                tables_list_resp = self.service.tables()\
+                    .list(projectId=project_id,
+                          datasetId=dataset_id,
+                          pageToken=tables_list_resp['nextPageToken'])\
+                    .execute()
+            # If there is no next page, then the table doesn't exist.
+            else:
+                # do insert
+                self.log.info('Creating Table %s:%s.%s',
+                              project_id, dataset_id, table_id)
+
+                try:
+                    self.service.tables().insert(
```

Comment:
what if you don't list before insert? trying to understand the value added piece of checking existence as opposed to blind insertion.